src/lightgbm/src/main/scala/LightGBMClassifier.scala

```diff
@@ -54,8 +54,10 @@ class LightGBMClassifier(override val uid: String)
      */
     val actualNumClasses = getNumClasses(dataset)
     val networkParams = NetworkParams(getDefaultListenPort, inetAddress, port)
+    val optionalWeightCol = if (hasWeightCol) Some(getWeightCol) else None
```

Comment:
```suggestion
    val optionalWeightCol = get(weightCol)
```

Or something like this

Reply:
done!